pkg/provider/aws/resources/eks.go

```diff
@@ -288,35 +289,75 @@ func NodeGroupName(networkPlacement string, instanceType string) string {
 	return nodeGroupSanitizer.Apply(fmt.Sprintf("%s_%s", networkPlacement, instanceType))
 }
 
-func createAddOns(cluster *EksCluster, provenance []core.AnnotationKey) []*kubernetes.HelmChart {
-	return []*kubernetes.HelmChart{
-		{
-			Name:          cluster.Name + `-metrics-server`,
-			Chart:         "metrics-server",
-			ConstructRefs: provenance,
-			ClustersProvider: core.IaCValue{
-				Resource: cluster.Kubeconfig,
-				Property: CLUSTER_PROVIDER_IAC_VALUE,
-			},
-			Repo: `https://kubernetes-sigs.github.io/metrics-server/`,
+func createAddOns(cluster *EksCluster, vpc *Vpc, provenance []core.AnnotationKey) []core.Resource {
+	clustersProvider := core.IaCValue{
+		Resource: cluster,
+		Property: CLUSTER_PROVIDER_IAC_VALUE,
+	}
+	return []core.Resource{
+		&kubernetes.HelmChart{
+			Name:             cluster.Name + `-metrics-server`,
+			Chart:            "metrics-server",
+			ConstructRefs:    provenance,
+			ClustersProvider: clustersProvider,
+			Repo:             `https://kubernetes-sigs.github.io/metrics-server/`,
 		},
-		{
-			Name:          cluster.Name + `-cert-manager`,
-			Chart:         `cert-manager`,
-			ConstructRefs: provenance,
-			ClustersProvider: core.IaCValue{
-				Resource: cluster.Kubeconfig,
-				Property: CLUSTER_PROVIDER_IAC_VALUE,
-			},
-			Repo:    `https://charts.jetstack.io`,
-			Version: `v1.10.0`,
+		&kubernetes.HelmChart{
+			Name:             cluster.Name + `-cert-manager`,
+			Chart:            `cert-manager`,
+			ConstructRefs:    provenance,
+			ClustersProvider: clustersProvider,
+			Repo:             `https://charts.jetstack.io`,
+			Version:          `v1.10.0`,
 			Values: map[string]any{
 				`installCRDs`: true,
 				`webhook`: map[string]any{
 					`timeoutSeconds`: 30,
 				},
 			},
 		},
+		&HelmChartAlbController{
+			HelmChart: kubernetes.HelmChart{
+
+				Name:             cluster.Name + `-alb-c`,
+				Chart:            `aws-load-balancer-controller`,
+				ConstructRefs:    provenance,
+				Repo:             `https://aws.github.io/eks-charts`,
+				Version:          `1.4.7`,
+				ClustersProvider: clustersProvider,
+				Namespace:        "default", // kube-system if on fargate, default otherwise
```

Comment:
i wouldnt delete these because well clean it up, but they wont do anything since all the values are hardcoded within the factory